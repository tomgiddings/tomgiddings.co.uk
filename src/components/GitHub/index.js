// @flow
import React from 'react';
import axios from 'axios';

class GitHub extends React.Component<*> {
	constructor(props) {
	  super(props);
	  this.state = {
      repos: [],
      loading: true
    };
	}

  componentDidMount() {
    axios.get(`https://api.github.com/users/${this.props.user}/repos?page=1&per_page=6&sort=updated`)
    .then((response) => {
      this.setState({
        repos: response.data,
        loading: false
      });
    })
    .catch((error) => {
      console.log(error);
      this.setState({ loading: false })
    });
  }

  render() {
    return (
      <section>
        {this.state.repos.length > 0 && <h3>Latest on GitHub</h3>}
        {this.state.loading && <div className="loading"><i className="fas fa-spinner fa-spin"></i>Loading...</div>}
        <ul className="cards">
          {this.state.repos.map(repo =>
            <li key={repo.id}>
              <a href={repo.html_url}>
                <h4>{repo.name}</h4>
                <p>{repo.description}</p>
              </a>
            </li>
          )}
        </ul>
      </section>
    );    
  }
}
export default GitHub;
