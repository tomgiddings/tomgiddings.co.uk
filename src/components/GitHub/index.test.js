import React from 'react';
import { render, cleanup } from '@testing-library/react';
import GitHub from './';

afterEach(cleanup);

it('Renders correctly', () => {
  const { asFragment } = render(<GitHub user="tomgiddings" />);
  expect(asFragment()).toMatchSnapshot();
});
