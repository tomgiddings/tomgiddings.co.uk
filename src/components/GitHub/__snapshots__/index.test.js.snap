// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<DocumentFragment>
  <section>
    <div
      class="loading"
    >
      <i
        class="fas fa-spinner fa-spin"
      />
      Loading...
    </div>
    <ul
      class="cards"
    />
  </section>
</DocumentFragment>
`;
